File: ff4j-spring-boot-starter/src/main/java/org/ff4j/spring/boot/domain/FlippingStrategyApiBean.java
package org.ff4j.spring.boot.domain;

import lombok.Getter;
import lombok.Setter;
import org.ff4j.core.FlippingStrategy;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

/**
 * Created by Paul
 *
 * @author <a href="mailto:paul58914080@gmail.com">Paul Williams</a>
 */
public class FlippingStrategyApiBean implements Serializable {

    private static final long serialVersionUID = 2257391205134600598L;

    @Getter
    @Setter
    private String type;

    @Getter
    @Setter
    private Map<String, String> initParams = new HashMap<>();

    public FlippingStrategyApiBean() {
        super();
    }

    public FlippingStrategyApiBean(FlippingStrategy fs) {
        this.type = fs.getClass().getCanonicalName();
        this.initParams = fs.getInitParams();
    }
}

File: ff4j-spring-services/src/main/kotlin/org/ff4j/services/domain/FlippingStrategyApiBean.kt
/*-
 * #%L
 * ff4j-spring-services
 * %%
 * Copyright (C) 2013 - 2023 FF4J
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package org.ff4j.services.domain

import org.ff4j.core.FlippingStrategy
import java.io.Serializable

/**
 * Created by Paul
 *
 * @author <a href="mailto:paul58914080@gmail.com">Paul Williams</a>
 */
class FlippingStrategyApiBean : Serializable {

  companion object {
    private const val serialVersionUID = 2257391205134600598L
  }

  var type: String? = null
  var initParams: MutableMap<String, String> = HashMap()

  constructor() : super()

  constructor(flippingStrategy: FlippingStrategy) {
    type = flippingStrategy.javaClass.canonicalName
    initParams = flippingStrategy.initParams
  }
}
